llvm/lib/Target/DirectX/DXILRootSignature.cpp

```diff
@@ -186,20 +187,23 @@ PreservedAnalyses RootSignatureAnalysisPrinter::run(Module &M,
   OS << "Root Signature Definitions"
      << "\n";
   uint8_t Space = 0;
-  for (const auto &P : RSDMap) {
-    const auto &[Function, RSD] = P;
-    OS << "Definition for '" << Function->getName() << "':\n";
+  for (const auto &F : M) {
+    const auto &RSD = RSDMap.find(&F);
```

Comment:
The type here is `SmallDenseMap<const Function *, mcdxbc::RootSignatureDesc>::iterator` - since this is actually a copy we shouldn't use a const reference here.
```suggestion
    auto It = RSDMap.find(&F);
```